src/ray/raylet/scheduling/cluster_task_manager.h

```diff
@@ -184,6 +226,10 @@ class ClusterTaskManager {
   /// Track the cumulative backlog of all workers requesting a lease to this raylet.
   std::unordered_map<SchedulingClass, int> backlog_tracker_;
 
+  WorkerPoolInterface &worker_pool_;
```

Comment:
Can we pass these as dependencies to `ScheduleAndDispatchTasks`, like we did before? It'd be nice to keep these out of the ClusterTaskManager since it is shared state with the NodeManager.

Reply:
- I've tried to use `worker_pool` as a method parameter, but several interfaces rely on it directly or indirectly, such as `QueueAndScheduleTask` / `TasksUnblocked` / `OnNodeResourceUsageUpdated`.
- How about modify it once we remove the legacy scheduler? (I will add a TODO inside there method.)